### DISCUSSION TITLE
Document the fact that `array/push` is variadic.

### DISCUSSION THREAD

**Discussion by pepe** [janet-lang/janet#1284] [Ideas]
I just found this on repl by accident:

```
Janet 1.30.0-43a6a70e linux/x64/gcc - '(doc)' for help
repl:1:> (array/push @[] :a :b)
@[:a :b]
repl:2:> array/push


    cfunction
    src/core/array.c on line 179, column 1

    (array/push arr x)
    
    Insert an element in the end of an array. Modifies the input array and returns it.
```

And I like it, so let's just document that `x` is actually a variadic argument.

**Comment by sogaiu** [comment]
Oh nice.  Had no idea.

**Reply by primo-ppcg** [reply]
Used, for example, in `interpose` :wink: 
https://github.com/janet-lang/janet/blob/a13aeaf9557250639018802baceeefb78b834846/src/boot/boot.janet#L1726-L1730

**Reply by pepe** [reply]
@primo-ppcg flies higher as always :-D.

**Comment by sogaiu** [comment]
@pepe So will you make a PR?

I guess at least [these lines](https://github.com/janet-lang/janet/blob/159c6129249c2d0b728433488353313a58531ca5/src/core/array.c#L180-L181) are involved:

```c
              "(array/push arr x)",
              "Insert an element in the end of an array. Modifies the input array and returns it.") {
```

Possibly of use is to look at what the docstring is for `buffer/push`:

```
    (buffer/push buffer & xs)

    Push both individual bytes and byte sequences to a buffer. For each 
    x in xs, push the byte if x is an integer, otherwise push the 
    bytesequence to the buffer. Thus, this function behaves like both 
    `buffer/push-string` and `buffer/push-byte`. Returns the modified 
    buffer. Will throw an error if the buffer overflows.
```

**Reply by pepe** [reply]
I should, but today I am not by the computer.
